mod/parse_url.php

```diff
@@ -25,10 +27,13 @@ function parse_url_content(App $a)
 
 	$br = "\n";
 
-	if (!empty($_GET['binurl'])) {
+	if (!empty($_GET['binurl']) && Strings::isHex($_GET['binurl'])) {
 		$url = trim(hex2bin($_GET['binurl']));
-	} else {
+	} elseif (!empty($_GET['url'])) {
 		$url = trim($_GET['url']);
+	// fallback in case no url is valid
+	} else {
+		$a->internalRedirect();
```

Comment:
Can you please redirect to `/parse_url` instead?

Reply:
We would create an infinite loop in this case.

Reply:
Ha, nevermind, it should just return empty string instead. it is supposed to be a `rawContent` endpoint, never to be accessed directly with a browser, so a redirection still is unwarranted.